app/src/main/java/org/luxoft/sdl_core/BluetoothHandler.java

```diff
@@ -47,6 +57,38 @@
             .fromString("00001102-0000-1000-8000-00805f9b34fb");
 
 
+    private String GenerateDisconnectMessage(BluetoothPeripheral peripheral) {
+
+        try {
+            JSONObject message = new JSONObject();
+            message.put(PARAM_ACTION, "ON_DEVICE_DISCONNECTED");
+
+            JSONObject params = new JSONObject();
+            params.put(PARAM_ADDRESS, peripheral.getAddress());
+            message.put(PARAMS, params);
+            return message.toString();
+        } catch (JSONException ex) {
+            Log.i(TAG, "ON_DEVICE_DISCONNECTED msg Failed", ex);
+        }
+        return "";
```

Comment:
```suggestion
        return null;
```

Reply:
fixed in 009c54dcc81215d2fa7d79cbfdb665850ed5309f